platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/builders/internal/AbstractTaskSchedulerBuilder.java

```diff
@@ -130,12 +130,30 @@ public AbstractTaskSchedulerBuilder<OUT> configure(@NonNull final TaskSchedulerC
         return this;
     }
 
+    /**
+     * Ensures that direct schedulers do not have an unhandled task capacity set.
+     *
+     * <p>If the scheduler type is {@link TaskSchedulerType#DIRECT} or {@link TaskSchedulerType#DIRECT_THREADSAFE}
+     * and the unhandled task capacity is not 1, an {@link IllegalArgumentException} is thrown.
+     *
+     * @param type the type of the task scheduler
+     * @param unhandledTaskCapacity the unhandled task capacity
+     * @throws IllegalArgumentException if the type is direct or direct threadsafe and the unhandled task capacity is not 1
+     */
+    public static void validateDirectScheduler(TaskSchedulerType type, long unhandledTaskCapacity) {
```

Comment:
generally, the convention we follow is to add `final` and `@NonNull`/`@Nullable` to parameters

Reply:
Updated